src/ekat/ekat_assert.hpp

```diff
@@ -27,13 +27,25 @@
     }                                                                   \
   } while(0)
 
-#define IMPL_KERNEL_THROW(condition, msg)                             \
-  do {                                                                \
-    if ( ! (condition) ) {                                            \
-      printf("KERNEL CHECK FAILED:\n   %s\n   %s\n",#condition,msg);  \
-      Kokkos::abort("");                                              \
-    }                                                                 \
+// SYCL cannot printf like the other backends quite yet
+#ifdef __SYCL_DEVICE_ONLY__
+#define IMPL_KERNEL_THROW(condition, msg)				\
+  do {									\
+    if ( ! (condition) ) {						\
+      const __attribute__((opencl_constant)) char format[] = "KERNEL CHECK FAILED:\n   %s\n"; \
```

Comment:
Since with non-SYCL we do print the condition as well, any reason not to print it for SYCL as well?

Reply:
The above attribute `const __attribute__((opencl_constant))` along with `sycl::ext::oneapi::experimental::printf` was only needed for print functionality from a SYCL device-side kernel and will be handled under the macro `__SYCL_DEVICE_ONLY__`. All the other device and host-side calls from SYCL will follow through the usual universal else case. 

This separate treatment will soon be resolved as soon as the standard printf gets supported for the kernel-side calls from llvm, and the above would not be needed any further

Reply:
What I'm saying is, can't you do
```
... format[] = "KERNEL CHECK FAILED:\n   %s\n   %s\n";
sycl::ext::oneapi::experimental::printf(format,#condition,msg);
```
?

Reply:
My bad, yes `#condition` can be included. I've just tested. Will make the changes.